File: 0110_javafx/gui020/Gui.java
// referred:
// http://libro.tuyano.com

import java.io.IOException;
import javafx.application.Application;
import javafx.fxml.FXMLLoader;
import javafx.scene.Scene;
import javafx.scene.layout.BorderPane;
import javafx.stage.Stage;

public class Gui extends Application {
  public static void main(String[] args) {
    launch(args);
  }

  @Override public void start(Stage stage) {
    BorderPane root;
    try {
      root = (BorderPane)FXMLLoader.load(getClass().getResource("gui.fxml"));
      Scene scene = new Scene(root, 200, 100);
      stage.setScene(scene);
      stage.show();
    } catch (IOException e) {
      e.printStackTrace();
    }
  }
}

File: 0110_javafx/gui020/Gui.clj
(ns Gui)

(import '[java.io IOException])
(import '[javafx.application Application])
(import '[javafx.fxml FXMLLoader])
(import '[javafx.scene Scene])
(import '[javafx.stage Stage])

;(require 'clojure.pprint)

(gen-class
  :name "Gui"
  :main true
  :extends javafx.application.Application)

(defn -start [this ^Stage stage]
  ;(clojure.pprint/pprint (. (. this getClass)
  ;                          getResource "gui.fxml"))
  (try
    (let [root (FXMLLoader/load (. (. this getClass)
                                   getResource "gui.fxml"))
          scene (Scene. root 200 100)]
      (.setScene stage scene)
      (.show stage))
    (catch IOException e (.printStackTrace e))
    ))

(defn -main [& args]
  (Application/launch (Class/forName "Gui") (into-array String [])))
